internal/codeConfiguration/task.ts

```diff
@@ -0,0 +1,48 @@
+import { execute } from "../execute";
+import { ParamSchema, JSParamValues, ParamKinds, ParamSchemaKind } from "./paramTypes";
+
+export type Params = Record<string, ParamKinds | ParamSchema>;
+
+export type ParamValues<PSchema extends Params> = {
+  [PSlug in keyof PSchema]: JSParamValues[ParamSchemaKind<PSchema[PSlug]>];
+};
+
+export type TaskConfig<TParams extends Params> = {
+  slug: string;
+  name: string;
+  description?: string;
+  parameters?: TParams;
+  requireRequests?: boolean;
+  allowSelfApprovals?: boolean;
+  timeout?: number;
+  constraints?: Record<string, string>;
+  runtime?: "standard" | "workflow";
+};
+
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export type UserFunc<TParams extends Params> = (params: ParamValues<TParams>) => any;
```

Comment:
Add a generic for the output so the output type is retained when a user calls a task function directly:

```suggestion
export type UserFunc<TParams extends Params, TOutput> = (params: ParamValues<TParams>) => TOutput;
```